### ISSUE TITLE
DeprecationWarning in Node 7 when using browserConsoleLogOptions.path.

### ISSUE THREAD

**Issue by chrahunt** [karma-runner/karma#2763]
### Expected behaviour

Executing test with config that includes `browserConsoleLogOptions` with `path` will successully (and quietly) write browser console to the given `path`.

### Actual behaviour

Each time browser console is written to `path`, a `DeprecationWarning` appears:

```
(node:3804) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
```

Running using `node --trace-warnings node_modules/.bin/karma` gives:

```
(node:12822) DeprecationWarning: Calling an asynchronous function without callback is deprecated.
    at maybeCallback (fs.js:98:42)
    at Object.fs.write (fs.js:729:14)
    at Server.<anonymous> (.../node_modules/karma/lib/server.js:231:10)
    at emitThree (events.js:121:20)
    at Server.emit (events.js:197:7)
    at Browser.onInfo (.../node_modules/karma/lib/browser.js:112:15)
    at Socket.<anonymous> (.../node_modules/karma/lib/events.js:13:22)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at .../node_modules/socket.io/lib/socket.js:503:12
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

The relevant line here is `Server.<anonymous> (.../node_modules/karma/lib/server.js:231:10)`. [The code in question](https://github.com/karma-runner/karma/blob/545dc05d11b295c71a7be39ddf0d09f2d71429b3/lib/server.js#L231).

### Environment Details

- Karma version (output of `karma --version`): Karma version: 1.7.0
- Relevant part of your `karma.config.js` file:

```js
browserConsoleLogOptions: {
  level: 'debug',
  path: 'karma-out.log',
  terminal: false
}
```

This is with node v7.10.0.

### Steps to reproduce the behaviour

```sh
git clone git@github.com:chrahunt/TagProReplays.git \
&& git checkout 5da3cb5                             \
&& npm test
```

Based on the [docs](https://nodejs.org/api/fs.html#fs_fs_write_fd_string_position_encoding_callback) we should be waiting for the callback between writes or using [`fs.createWriteStream`](https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options).

**Comment by chrahunt** [comment]
Blocked on #2765 since those are the relevant tests for the changes needed here.

**Comment by chrahunt** [comment]
Fixed by #2939.
